coderd/httpmw/apikey.go

```diff
@@ -31,6 +31,19 @@ func APIKey(r *http.Request) database.APIKey {
 	return apiKey
 }
 
+// User roles are the 'subject' field of Authorize()
+type userRolesKey struct{}
+
+// UserAuthorizationRoles returns the roles used for authorization.
+// Comes from the ExtractAPIKey handler.
+func UserAuthorizationRoles(r *http.Request) database.GetAuthorizationUserRolesRow {
```

Comment:
nit: `httpmw.UserAuthorizationRoles` vs `database.GetAuthorizationUserRoles` -- pick one or othe other, else it'll be super annoying to find later!

Reply:
:+1: